lake_hylia.logic

```diff
@@ -48,7 +48,9 @@ area LakeHylia:
       entrance DigCaveFusion
     node DigCaveFusionExit:
       exit DigCaveFusion
-    DigCaveFusionExit -> DigCaveFusionEntrance: flag HyliaDigCaveFusion
+    node DigCaveFusionIsland
+    DigCaveFusionExit -> DigCaveFusionIsland: flag HyliaDigCaveFusion
+    DigCaveFusionIsland -> DigCaveFusionEntrance: ( flag HyliaDigCaveFusion | item MoleMitts )
```

Comment:
The flag should always be active if we have reached the island, no need to check it again here.
```suggestion
    DigCaveFusionIsland -> DigCaveFusionEntrance: item MoleMitts
```